### PR TITLE
implement apex gateway ListenAndServe internally

### PR DESCRIPTION
**Pull request by yusuf-erdem**

### CODE REVIEW

File: wrappers/apexgateway/thundra_apex_gateway.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	lambda.Start(thundra.Wrap(wrapper(h)))
+
+	return nil
+}
+
+func wrapper(h http.Handler) func(ctx context.Context, e events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+	return func(ctx context.Context, e events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+		r, err := gateway.NewRequest(ctx, e)
+		if err != nil {
+			return events.APIGatewayProxyResponse{}, err
```
Comment (serkan-ozal) [inline wrappers/apexgateway/thundra_apex_gateway.go:27]: shouldn't it be 
```go
return nil, err
```
?

File: wrappers/apexgateway/thundra_apex_gateway.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"context"
+	"github.com/apex/gateway"
+	"github.com/thundra-io/thundra-lambda-agent-go/v2/thundra"
+	"net/http"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+func ListenAndServe(h http.Handler) error {
```
Comment (serkan-ozal) [inline wrappers/apexgateway/thundra_apex_gateway.go:13]: what about another `ListenAndServe` and `wrapper` for API Gateway v2 (https://github.com/apex/gateway/blob/master/v2/gateway.go)?

Comment (yusuf-erdem) [reply wrappers/apexgateway/thundra_apex_gateway.go:13]: package apexgatewayv2 added for this requirement

File: wrappers/apexgateway/thundra_apex_gateway.go (hunk 1 above)
Comment (yusuf-erdem) [reply wrappers/apexgateway/thundra_apex_gateway.go:27]: according to net/http package ListenAndServe function, they comment out that 
**// ListenAndServe always returns a non-nil error.**,  thats why this function returns empty response.
What is the reason of thinking it should return nil.

**Review by serkan-ozal — APPROVED** [review]

**Pull request merged**